config/ompi_check_ucx.m4

```diff
@@ -16,65 +19,72 @@
-                 [ompi_check_ucx_libdir="$with_ucx_libdir"])
+	AS_IF([test "$with_ucx" != "no"],
+              [AS_IF([test -n "$with_ucx" && test "$with_ucx" != "yes"],
+                     [ompi_check_ucx_dir="$with_ucx"
+                      files=`ls $ompi_check_ucx_dir/lib64/libucp.* 2> /dev/null | wc -l`
+                      AS_IF([test "$files" -gt 0],
+                            [ompi_check_ucx_libdir=$ompi_check_ucx_dir/lib64],
+                            [ompi_check_ucx_libdir=$ompi_check_ucx_dir/lib])])
+               AS_IF([test -n "$with_ucx_libdir" && test "$with_ucx_libdir" != "yes"],
+                     [ompi_check_ucx_libdir="$with_ucx_libdir"])
 
-           ompi_check_ucx_extra_libs="-L$ompi_check_ucx_libdir"
+               OPAL_CHECK_PACKAGE([ompi_check_ucx],
+				  [ucp/api/ucp.h],
+				  [ucp],
+				  [ucp_cleanup],
+				  [],
+				  [$ompi_check_ucx_dir],
+				  [$ompi_check_ucx_libdir],
+				  [ompi_check_ucx_happy="yes"],
+				  [ompi_check_ucx_happy="no"])],
+              [ompi_check_ucx_happy="no"])
 
-           OPAL_CHECK_PACKAGE([$1],
-                              [ucp/api/ucp.h],
-                              [ucp],
-                              [ucp_cleanup],
-                              [$ompi_check_ucx_extra_libs],
-                              [$ompi_check_ucx_dir],
-                              [$ompi_check_ucx_libdir],
-                              [ompi_check_ucx_happy="yes"],
-                              [ompi_check_ucx_happy="no"])],
-          [ompi_check_ucx_happy="no"])
 
 
+	CPPFLAGS="$ompi_check_ucx_$1_save_CPPFLAGS"
+	LDFLAGS="$ompi_check_ucx_$1_save_LDFLAGS"
+	LIBS="$ompi_check_ucx_$1_save_LIBS"
 
-    CPPFLAGS="$ompi_check_ucx_$1_save_CPPFLAGS"
-    LDFLAGS="$ompi_check_ucx_$1_save_LDFLAGS"
-    LIBS="$ompi_check_ucx_$1_save_LIBS"
-
-    AC_MSG_CHECKING(for UCX version compatibility)
-    AC_REQUIRE_CPP
-    old_CPPFLAGS="$CPPFLAGS"
-    CPPFLAGS="$CPPFLAGS -I$ompi_check_ucx_dir/include"
-    AC_COMPILE_IFELSE(
+	AC_MSG_CHECKING(for UCX version compatibility)
+	AC_REQUIRE_CPP
+	old_CPPFLAGS="$CPPFLAGS"
+        AS_IF([test -n "$ompi_check_ucx_dir"],
+              [CPPFLAGS="$CPPFLAGS -I$ompi_check_ucx_dir/include"])
+	AC_COMPILE_IFELSE(
             [AC_LANG_PROGRAM([[#include <uct/api/version.h>]],
-                [[
-                ]])],
+			     [[
+			     ]])],
             [ompi_ucx_version_ok="yes"],
             [ompi_ucx_version_ok="no"])
 
-    AC_MSG_RESULT([$ompi_ucx_version_ok])
-    CPPFLAGS=$old_CPPFLAGS
+	AC_MSG_RESULT([$ompi_ucx_version_ok])
+	CPPFLAGS=$old_CPPFLAGS
+
+	AS_IF([test "$ompi_ucx_version_ok" = "no"], [ompi_check_ucx_happy="no"])
+
+    fi
 
-    AS_IF([test "$ompi_ucx_version_ok" = "no"], [ompi_check_ucx_happy="no"])
 
     AS_IF([test "$ompi_check_ucx_happy" = "yes"],
-          [$2],
+          [$1_CPPFLAGS="[$]$1_CPPFLAGS $ompi_check_ucx_CPPFLAGS"
+	   $1_LDFLAGS="[$]$1_LDFLAGS $ompi_check_ucx_LDFLAGS"
```

Comment:
Watch tabs vs. spaces here.

Open MPI is allergic to tabs.  😉

Reply:
will fix if it works

Reply:
@jsquyres btw, I copied this file from 2.x branch, it means that all release branches have tab/space mix :)